core/src/main/java/org/infinispan/util/functional/MarshallableFunctionalInterfaces.java

```diff
@@ -0,0 +1,341 @@
+import org.infinispan.commons.marshall.SerializeValueMatcher;
+import org.infinispan.commons.marshall.SerializeWith;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public final class MarshallableFunctionalInterfaces {
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, V> setValueReturnPrevOrNull() {
+      return SetValueReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, V> setValueIfAbsentReturnPrevOrNull() {
+      return SetValueIfAbsentReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> setValueIfAbsentReturnBoolean() {
+      return SetValueIfAbsentReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, V> setValueIfPresentReturnPrevOrNull() {
+      return SetValueIfPresentReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> setValueIfPresentReturnBoolean() {
+      return SetValueIfPresentReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> setValueIfEqualsReturnBoolean(V oldValue) {
+      return new SetValueIfEqualsReturnBoolean<>(oldValue);
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, V> removeReturnPrevOrNull() {
+      return RemoveReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, Boolean> removeReturnBoolean() {
+      return RemoveReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> removeIfValueEqualsReturnBoolean() {
+      return RemoveIfValueEqualsReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <V> BiConsumer<V, WriteEntryView<V>> setValueConsumer() {
+      return SetValue.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <V> Consumer<WriteEntryView<V>> removeConsumer() {
+      return Remove.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, Optional<V>> findReadWrite() {
+      return FindReadWrite.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, V> getReadWrite() {
+      return GetReadWrite.INSTANCE;
+   }
+
+   @SerializeWith(value = SetValueReturnPrevOrNull.Externalizer0.class,
```

Comment:
Yeah, this could be done with advanced externalizers and since they're constants I can pre-register them. Using advanced externalizers is recommended. At the start I wasn't sure if I could pre-register them and hence why I used `@SerializeWith`. You could still have a value matcher in advanced externalizer, you just have to add the method to AdvancedExternalizer